### PR TITLE
#5368 Remove trailing hyphens on signature

### LINKED ISSUE #5368: Don't add "--" separator if user signature begins with "--"
FlowCrypt keeps adding "--" at the beginning of my signature. Which is unfortunate, because my signature already begins with "-- ". It's pretty ugly.

I believe correct behavior would be to not add anything to the signature?

### PR DESCRIPTION
**Pull request by martgil**
This PR removes the trailing slash from the email signature.

close #5368 

----------------------------------

**Tests** _(delete all except exactly one)_:
- Does not need tests (refactor only, docs or internal changes)

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities
- [ ] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

File: extension/chrome/elements/compose-modules/compose-footer-module.ts [hunk 1]
```diff
@@ -33,14 +33,16 @@ export class ComposeFooterModule extends ViewModule<ComposeView> {
...
+    let sanitizedHtmlFooter = sanitizedPlainFooter.replace(/\n/g, '<br>');
     const footerFirstLine = sanitizedPlainFooter.split('\n')[0];
     if (!footerFirstLine) {
       return '';
     }
     if (/^[*\-_=+#~ ]+$/.test(footerFirstLine)) {
       return sanitizedHtmlFooter; // first line of footer is already a footer separator, made of special characters
     }
+    sanitizedHtmlFooter =
+      sanitizedHtmlFooter.startsWith('&nbsp;--') || sanitizedHtmlFooter.startsWith('--') ? sanitizedHtmlFooter.replace('--', '') : sanitizedHtmlFooter;
```
Comment (sosnovsky) [inline extension/chrome/elements/compose-modules/compose-footer-module.ts:45]: Looks like we already checking if footer starts with separator at line 41, but for some reason this regex didn't work in user's case.

Let's just remove `&nbsp;` from `footerFirstLine`, so it'll look like:
```js
const footerFirstLine = sanitizedPlainFooter.split('\n')[0]?.replace('&nbsp;', '');
```

Comment (martgil) [reply extension/chrome/elements/compose-modules/compose-footer-module.ts:45]: @sosnovsky thank you very much. I have tested the updated code and the regex works after the removal of the `&nbsp;` and I get the desired result.

Comment (sosnovsky) [reply extension/chrome/elements/compose-modules/compose-footer-module.ts:45]: Were you able to reproduce original issue reported by user? As for me it works well even with additional newline on the beginning of signature

Comment (martgil) [reply extension/chrome/elements/compose-modules/compose-footer-module.ts]: Yes, the new line doesn't seem to interfere with my testing case as well. Perhaps there is a non-breaking space lying around somewhere there?

Comment (sosnovsky) [reply extension/chrome/elements/compose-modules/compose-footer-module.ts]: On screenshots it looks like plain signature, without any non-standard characters

**Pull request closed without merging**